### DISCUSSION TITLE
Filtering Using bash

### DISCUSSION THREAD

**Discussion by orisa** [supabase/supabase#8207] [Questions] [answered]
Goodday. I'm currently using bash for requests. I'm trying to filter a product table where a colors column of type json equals a specific json.

Here is a sample of the GET request:
https://myDatabaseURL/rest/v1/product?colors=eq.{"color1":"bcb2b41e-4009-46b5-b4d9-38a78409152e","color2":"82f1b68d-9a23-4c86-8eeb-1f5756d4ffa9"}&select=*

error response:
{
    "message": "operator does not exist: json = unknown",
    "code": "42883",
    "details": null,
    "hint": "No operator matches the given name and argument types. You might need to add explicit type casts."
}

How do I accomplish this using bash? Thanks.

**Comment by steve-chavez** [comment, marked as answer]
@orisa Can you define your column as `jsonb` instead of `json`?

Equality is not defined for the `json` type, works for `jsonb` though:

```sql
select '{}'::json = '{}'::json;
ERROR:  operator does not exist: json = json

select '{}'::jsonb = '{}'::jsonb;
 ?column?
----------
 t
```

**Reply by orisa** [reply]
Alright let me give it a shot.

**Reply by orisa** [reply]
@steve-chavez Apologies for the late response. I've quite busy. I tried it and it works but I am unable to set a default value for the column using the table editor. Here is the default value I tried: '{"size1":"","size2":"","size3":"","size4":"","size5":""}'::jsonb . The loader save button just kept spinning.

**Reply by steve-chavez** [reply]
@orisa Try with plain SQL:

```
ALTER TABLE tbl_name ALTER COLUMN col_name TYPE jsonb USING col_name::jsonb;
```

About the Table Editor, that sounds like a bug, I suggest you report it in https://github.com/supabase/supabase/issues.

**Reply by orisa** [reply]
Thanks. I will give this a shot.

**Reply by orisa** [reply]
Setting a jsonb column to a default json value such as {"color1": "", "color2": "", "color3": "", "color4": "", "color5": ""} now works in the editor. Thanks
